lua/telescope/pickers/entry_display.lua

```diff
@@ -3,6 +3,62 @@ local strings = require "plenary.strings"
 local entry_display = {}
 entry_display.truncate = strings.truncate
 
+---@tag telescope.pickers.entry_display
+
+---@brief [[
+--- Entry Display is used to format each entry shown in the result panel.
+---
+--- Entry Display create() will give us a function based on the configuration
+--- of column widths we pass into to it. We then can use this function n times 
+--- to return a string based on structured input.
+---
+--- The create function will use the column widths passed to it in
+--- configaration.items. Each item in that table is the number of characters in
+--- the column. It's also possible for the final column to not have a fixed width,
+--- this will be shown in the configuartion as 'remaining = true'.
+---
+--- An example of this configuration is shown for the buffers picker
+---
+---   local displayer = entry_display.create {
+---     separator = " ",
+---     items = {
+---       { width = opts.bufnr_width },
+---       { width = 4 },
+---       { width = icon_width },
+---       { remaining = true },
+---     },
+---   }
+---
+--- This shows 4 columns, the first is defined in the opts as the width we'll use when display_string
+--- the number of the buffer. The second has a fixed width of 4 and the 3rd column's widht will be 
+--- decided by the width of the icons we use. The fourth column will use the remaining space.
+--- Finally we have also defined the seperator between each column will be the space " ".
+---
+---
+--- It's best practice for pickers to store the function returned by create() in a reference
+--- called display. This function is called for every entry which is displayed and passes in the
```

Comment:
We dont care how the reference is named. The important part is that its stored outside of the function reference which is passed to `display`, because `display` is called basically inside a loop. And creating it multiple time is a waste